tests/test_dimension.py

```diff
@@ -831,6 +831,29 @@ def test_implicit_dims(self):
 
         assert np.all(f.data == F)
 
+    def test_stepping_dim_in_condition_lowering(self):
+        """
+        Check that the compiler performs the following lowering::
+
+        if f[t, x] > 2 {            if f[time, x] > 2 {
```

Comment:
not sure I follow the docstring with the code below

Reply:
- the function is called `g` not `f`
- why is there two if
- why is it ` > 2` when the test is `<10`

This docstring makes it really hard to follwo what this test is about

Reply:
ok, did not realize you were matching the docs absolutely.
- Matched docstring
- If before transformation, if after XSubs
- Matched docstring